src/main/java/ch/njol/skript/effects/EffChange.java

```diff
@@ -288,7 +282,7 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 						hintManager.add(variable, hints);
 					}
 				}
-				if (!variable.isLocal()) {
+				if (!variable.isLocal() && !variable.getName().toString(ContextlessEvent.get()).startsWith(Variables.EPHEMERAL_VARIABLE_PREFIX)) {
```

Comment:
Can ephemeral variables contain expressions? Are we sure it's safe to evaluate the name with the contextless event?

Reply:
it's not, see above comment